### PR TITLE
LPD-33712 Increase z-index for maximized editor

### PR DESCRIPTION
**Pull request by fortunatomaldonado**
Requries https://github.com/liferay-frontend/liferay-portal/pull/4401

https://liferay.atlassian.net/browse/LPP-55194
https://liferay.atlassian.net/browse/LPD-33712

With the update of CKEditor v.4.21.0-liferay.4, the following commits resolves the stylescombo not being visible when editor is maximized.

Please let me know if there are any questions or comments about this.
Thank you!

### CODE REVIEW

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/test/playwright/tests/frontend-editor-ckeditor-sample-web/stylesCombo.spec.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		await page.goto(layout.friendlyURL);
+
+		const editButton = page.getByRole('link', {name: 'Edit'});
+
+		await editButton.waitFor({state: 'visible'});
+		await editButton.click();
+	});
+
+	await test.step('Add sample ckeditor portlet on page ', async () => {
+		const searchFragment = page.getByLabel('Search Fragments and Widgets');
```
Comment (dsanz) [inline modules/test/playwright/tests/frontend-editor-ckeditor-sample-web/stylesCombo.spec.ts:28]: probably we can reduce the amount of code here if we reuse some primitive to add a portlet to a page, pls find occurrences of `addPortlet`  in pw tests

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/css/main.scss [hunk 2]
```diff
@@ -263,6 +263,10 @@
 	z-index: 100 !important;
 }
 
+.cke_combopanel.lfr-maximized {
```
Comment (dsanz) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/css/main.scss:266]: do we know why this stopped working?

Comment (fortunatomaldonado) [reply modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/css/main.scss:266]: I added `lfr-maximized` through liferay-ckeditor here: https://github.com/liferay/liferay-ckeditor/commit/62fe441be4c82545b29a0eb8342139d4613830f1

It checks if the editor is maximized and add or remove the css accordingly.
Please let me know if I answered your question.

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/test/playwright/tests/frontend-editor-ckeditor-sample-web/stylesCombo.spec.ts (hunk 1 above)
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-editor-ckeditor-sample-web/stylesCombo.spec.ts]: Hello @dsanz 
I made the changes! 
Let me know if I should change anything else.
Thank you!

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/test/playwright/playwright.config.ts [hunk 3]
```diff
@@ -28,7 +28,7 @@ import {config as featureFlagWebConfig} from './tests/feature-flag-web/config';
 import {config as fragmentWebConfig} from './tests/fragment-web/config';
 import {config as frontendDataSetAdminWebConfig} from './tests/frontend-data-set-admin-web/config';
 import {config as frontendDataSetWebConfig} from './tests/frontend-data-set-web/config';
-import {config as frontendEditorCKEditorSampleWebConfig} from './tests/frontend-editor-ckeditor-sample-web/config';
+import {config as frontendCKEditorSampleWebConfig} from './tests/frontend-ckeditor-sample-web/config';
```
Comment (markocikos) [inline modules/test/playwright/playwright.config.ts:31]: Why this change of project name? PW project name should normally match module name, [frontend-editor-ckeditor-sample-web](https://github.com/liferay/liferay-portal/tree/master/modules/apps/frontend-editor/frontend-editor-ckeditor-sample-web). There is an argument to be had if it should be `frontend-editor-ckeditor-web` (no `sample`), but `frontend-ckeditor-sample-web` seems just wrong.

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts
Comment (markocikos) [inline modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:1]: As this test is on `Classic` variant, we should call this file `classic.spec.ts`. The idea is to group tests by editor version. We already have `legacy.spec.ts` that contains test on the legacy version.

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+export const test = mergeTests(
+	apiHelpersTest,
+	featureFlagsTest({
+		'LPS-178052': true,
+	}),
+	isolatedSiteTest,
+	loginTest()
+);
+
+test('@LPD-33712 stylesCombo dropdown visible with maximized', async ({
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:26]: You can define tag more formally, in a separate argument, like we do for example [here](https://github.com/liferay/liferay-portal/blob/master/modules/test/playwright/tests/frontend-data-set-admin-web/tests/data-set-admin/creationActions.spec.ts#L55).

To avoid description being too technical (plugin name), we can say something like `Dropdown menus are visible when maximized`

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	await test.step('Select Maximized button and check stylesCombo dropdown', async () => {
+		await page.goto(
+			`${liferayConfig.environment.baseUrl}/web${site.friendlyUrlPath}${layout.friendlyUrlPath}`
+		);
+
+		const ckeditorSampleWidgetClassicTab = page
+			.getByRole('tab', {
+				name: 'Classic',
+			})
+			.first();
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:56]: Why `first`? There should be only one `Classic` tab.

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+		await ckeditorSampleWidgetClassicTab.waitFor({state: 'visible'});
+		await ckeditorSampleWidgetClassicTab.click();
+
+		const maximizedButton = page.getByLabel('Maximize').first();
```
Comment (markocikos) [inline modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:61]: Why first? There should be only one `Maximize` tab. Maybe we can be more specific with something like
```js
.getByRole('button', {
    name: 'Maximize',
})
```

**Review by markocikos — CHANGES_REQUESTED** [review]
@fortunatomaldonado please see comments inline

File: modules/test/playwright/playwright.config.ts (hunk 3 above)
Comment (fortunatomaldonado) [reply modules/test/playwright/playwright.config.ts:31]: Hello @markocikos 
Having the same name, caused issues for CI as it had could not find frontend-editor-ckeditor-sample-web. Changing the name, made the module deploy correctly and pass the PW tests.

Here is a link about the situation.
https://liferay.slack.com/archives/C19PWQ74J/p1727287797591529

Let me know if you want things to be changed.

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:1]: Got it!

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts (hunk 4 above)
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:26]: Thanks!

File: modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts (hunk 5 above)
Comment (fortunatomaldonado) [reply modules/test/playwright/tests/frontend-ckeditor-sample-web/stylesCombo.spec.ts:56]: This is no longer needed with the way this test was created. Removed.

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

File: modules/test/playwright/playwright.config.ts (hunk 3 above)
Comment (markocikos) [reply modules/test/playwright/playwright.config.ts]: @fortunatomaldonado If I understand the problem correctly, this definitely not the proper solution. If a module cannot be found, how can renaming to a non-existing module possibly work? Something very hacky is going on here. The solution is to fix module deployment in `frontend-editor-ckeditor-sample-web`, to use the correct configuration format, and add the new test to it.

Comment (markocikos) [reply modules/test/playwright/playwright.config.ts]: I pulled your PR and and the tests are locally not even showing up in framework. That's why CI is green, tests are not run at all.

image

Comment (dsanz) [reply modules/test/playwright/playwright.config.ts]: Origin of this change is in this [slack conversation](https://liferay.slack.com/archives/C19PWQ74J/p1727287797591529). Let's keep talking there, this looks a bit odd to me as well

Comment (dsanz) [reply modules/test/playwright/playwright.config.ts]: probably, removing the `-sample` part in the PW project folder will work 
That's how we arranged tests in `frontend-data-set-web` pw folder. There we deploy `frontend-data-set-sample-web` 

So, pw folder here should be `frontend-editor-ckeditor-web` and there we'll deploy the sample-web module

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:sf

**Comment by fortunatomaldonado** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
@fortunatomaldonado LGTM! :sparkles: 

image

File: modules/apps/frontend-editor/frontend-editor-ckeditor-web/test.properties [hunk 7]
```diff
@@ -2,7 +2,7 @@
 ## Playwright
 ##
 
-    playwright.test.project[playwright-js-tomcat90-mysql57-jdk8][relevant][frontend-editor-rule]=frontend-editor-ckeditor-sample-web
+    playwright.test.project[playwright-js-tomcat90-mysql57-jdk8][relevant][frontend-editor-rule]=frontend-editor-ckeditor-web
```
Comment (dsanz) [inline modules/apps/frontend-editor/frontend-editor-ckeditor-web/test.properties:5]: nice!

**Comment by fortunatomaldonado** [comment]
ci:forward
